.github/workflows/artifacts.yml

```diff
@@ -0,0 +1,83 @@
+name: Artifacts generation and upload
+on:
+  workflow_run:
+    workflows: ["Mattermost Build"]
+    types:
+      - completed
+jobs:
+  upload-s3:
+    name: cd/Upload artifacts to S3
+    runs-on: ubuntu-22.04
+    env:
+      REPO_NAME: ${{ github.event.repository.name }}
+    if: >
+      github.event.workflow_run.event == 'pull_request' &&
+      github.event.workflow_run.conclusion == 'success'
+    steps:
+      - name: cd/Configure AWS
+        uses: aws-actions/configure-aws-credentials@07c2f971bac433df982ccc261983ae443861db49 # v1-node16
+        with:
+          aws-region: us-east-1
+          aws-access-key-id: ${{ secrets.PR_BUILDS_BUCKET_AWS_ACCESS_KEY_ID }}
+          aws-secret-access-key: ${{ secrets.PR_BUILDS_BUCKET_AWS_SECRET_ACCESS_KEY }}
+      - name: cd/Download artifacts
+        uses: dawidd6/action-download-artifact@0c49384d39ceb023b8040f480a25596fd6cf441b # v2.26.0
+        with:
+          workflow_conclusion: success
+          name: server-dist-artifact
+      # Get Branch name from calling workflow
+      # Search for the string "pull" and replace it with "PR" in branch-name
+      - name: cd/Get branch name
+        run: echo "BRANCH_NAME=$(echo ${{ github.event.workflow_run.head_branch }} | sed 's/^pull\//PR-/g')" >> $GITHUB_ENV
+      - name: cd/Upload artifacts to S3
+        run: |
+          aws s3 cp ${{github.workspace}}/ s3://pr-builds.mattermost.com/$REPO_NAME/$BRANCH_NAME/ --acl public-read --cache-control "no-cache" --recursive --no-progress
+          aws s3 cp ${{github.workspace}}/ s3://pr-builds.mattermost.com/$REPO_NAME/commit/${{ github.sha }}/ --acl public-read --cache-control "no-cache" --recursive --no-progress
```

Comment:
Same here . Better use the path instead of uploading the whole workspace . You never know what files are downloaded or using by other actions

Reply:
Fixed with https://github.com/mattermost/mattermost-server/compare/a6f7e73e5165d1261f4e55dd04b50e7a67cecfc0..4a6c1d500f8fa1ff3a935f4a5b52be3f23df19c5